web/docs/auth/username-and-pass.md

```diff
@@ -1,3 +1,659 @@
+
+  If you need to add extra fields to the user entity, we suggest doing it in a separate step after the user logs in for the first time.
+  :::
+
+You can use it like this:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```jsx title="client/pages/auth.jsx"
+// Importing the signup and login actions 👇
+import signup from '@wasp/auth/signup'
+import login from '@wasp/auth/login'
+
+import { useState } from 'react'
+import { useHistory } from 'react-router-dom'
+import { Link } from 'react-router-dom'
+
+export function Signup() {
+  const [username, setUsername] = useState('')
+  const [password, setPassword] = useState('')
+  const [error, setError] = useState(null)
+  const history = useHistory()
+
+  async function handleSubmit(event) {
+    event.preventDefault()
+    try {
+      await signup({
+        username,
+        password,
+      })
+      await login(username, password)
+      history.push("/")
+    } catch (error) {
+      setError(error)
+    }
+  }
+
+  return (
+    <form onSubmit={handleSubmit}>
+      {/* ... */}
+    </form>
+  );
+}
+```
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```tsx title="client/pages/auth.tsx"
+// Importing the signup and login actions 👇
+import signup from '@wasp/auth/signup'
+import login from '@wasp/auth/login'
+
+import { useState } from 'react'
+import { useHistory } from 'react-router-dom'
+import { Link } from 'react-router-dom'
+
+export function Signup() {
+  const [username, setUsername] = useState('')
+  const [password, setPassword] = useState('')
+  const [error, setError] = useState<Error | null>(null)
+  const history = useHistory()
+
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault()
+    try {
+      await signup({
+        username,
+        password,
+      })
+      await login(username, password)
+      history.push("/")
+    } catch (error: unknown) {
+      setError(error as Error)
+    }
+  }
+
+  return (
+    <form onSubmit={handleSubmit}>
+      {/* ... */}
+    </form>
+  );
+}
+```
+</TabItem>
+</Tabs>
+
+### 2. Creating your custom actions
+
+The code of your custom sign-up action can look like this:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```wasp title="main.wasp"
+// ...
+
+action signupUser {
+  fn: import { signUp } from "@server/auth/signup.js",
+  entities: [User]
+}
+```
+
+
+```js title="src/server/auth/signup.js"
+export const signUp = async (args, context) => {
+  // Your custom code before sign-up.
+  // ...
+
+  const newUser = context.entities.User.create({
+    data: {
+      username: args.username,
+      password: args.password // password hashed automatically by Wasp! 🐝
+    }
+  })
+
+  // Your custom code after sign-up.
+  // ...
+  return newUser
+}
+```
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```wasp title="main.wasp"
+// ...
+
+action signup {
```

Comment:
Action name isn't consistent with the name in the JS tab, and the TS code imports the action type as if it was named `signupUser`.

Reply:
Renamed 👍